lib/main.js

```diff
@@ -55,14 +67,56 @@ export default {
           });
         }
       }),
-    );
-
-    this.subscriptions.add(
       atom.config.observe('linter-tslint.useLocalTslint', (use) => {
         tslintCache.clear();
         this.useLocalTslint = use;
       }),
     );
+
+    atom.workspace.observeActivePaneItem((activeItem) => {
+      if (activeItem && activeItem.getGrammar) {
+        if (isTypescriptGrammar(activeItem.getGrammar())) {
+          activeItem.element.classList.add(editorClass);
+        }
+      }
+    });
+
+    // Command subscriptions
+    this.subscriptions.add(
+      atom.commands.add(`atom-text-editor.${editorClass}`, {
+        'linter-tslint:fix-file': () => {
+          const textEditor = atom.workspace.getActiveTextEditor();
+          // const filePath = textEditor.getPath();
+          // const text = textEditor.getText();
+
+          if (!textEditor || textEditor.isModified()) {
+            // Abort for invalid or unsaved text editors
+            atom.notifications.addError('Linter-TSLint: Please save before fixing');
+            return Promise.resolve();
```

Comment:
Just mark the function as `async` [here](https://github.com/AtomLinter/linter-tslint/pull/149/files#diff-16b7d75b70953cbf4755508170f2f24cR87) and all return values will be Promises. You can also use `await` on the `lint()` call, as well as getting rid of the chained `then()` calls.